test/Npgsql.Benchmarks/TypeHandlers/TypeHandlerBenchmarks.cs

```diff
@@ -35,29 +35,36 @@ class EndlessStream : Stream
             public override void Write(byte[] buffer, int offset, int count) { }
         }
 
-        readonly EndlessStream _stream = new EndlessStream();
-        readonly NumericHandler _handler = new NumericHandler();
+        readonly EndlessStream _stream;
+        readonly NpgsqlTypeHandler<T> _handler;
         readonly NpgsqlReadBuffer _readBuffer;
         readonly NpgsqlWriteBuffer _writeBuffer;
-        decimal _value;
+        T _value;
         int _elementSize;
 
-        public Numeric()
+        protected TypeHandlerBenchmarks(NpgsqlTypeHandler<T> handler)
         {
-            _stream = new EndlessStream(); _handler = new NumericHandler();
+            _stream = new EndlessStream();
+            _handler = handler ?? throw new ArgumentNullException(nameof(handler));
             _readBuffer = new NpgsqlReadBuffer(null, _stream, NpgsqlReadBuffer.MinimumSize, Encoding.UTF8);
             _writeBuffer = new NpgsqlWriteBuffer(null, _stream, NpgsqlWriteBuffer.MinimumSize, Encoding.UTF8);
         }
 
+        public IEnumerable<T> Values() => ValuesOverride();
+
+        protected virtual IEnumerable<T> ValuesOverride() => new[] { default(T) };
+
         [ParamsSource(nameof(Values))]
-        public decimal Value
+        public T Value
         {
             get => _value;
             set
             {
+                NpgsqlLengthCache cache = null;
+
                 _value = value;
-                _elementSize = _handler.ValidateAndGetLength(value, null);
-                _handler.Write(_value, _writeBuffer, null);
+                _elementSize = _handler.ValidateAndGetLength<T>(value, ref cache, null);
+                _handler.WriteWithLengthInternal(_value, _writeBuffer, null, null, false);
```

Comment:
Any reason for calling `WriteWithLengthInternal` instead of `Write`?

Reply:
See the comment above.